src/lib/es2015.iterable.d.ts

```diff
@@ -8,15 +8,23 @@ interface SymbolConstructor {
     readonly iterator: symbol;
 }
 
-interface IteratorResult<T> {
-    done: boolean;
-    value: T;
+interface IteratorYieldResult<TYield> {
+    done?: false;
+    value: TYield;
 }
 
-interface Iterator<T> {
-    next(value?: any): IteratorResult<T>;
-    return?(value?: any): IteratorResult<T>;
-    throw?(e?: any): IteratorResult<T>;
+interface IteratorReturnResult<TReturn> {
+    done: true;
+    value: TReturn;
+}
+
+type IteratorResult<T, TReturn = any> = IteratorYieldResult<T> | IteratorReturnResult<TReturn>;
+
+interface Iterator<T, TReturn = any, TNext = undefined> {

```

Comment:
can other code besides generators produces Iterators with different types for TReturn and TNext?
A 1-parameter Iterator type would be nicer to print. It probably doesn't matter much because it looks like Array (and other built-ins?) use IterableIterator, which is still 1-parameter.

Reply:
Unfortunately it is necessary for `Iterator` to support three parameters so that the correct assignment compatibility rules for `Generator` can be applied to it in certain situations.